app/javascript/components/miq-toolbar.jsx

```diff
@@ -66,7 +66,7 @@ const onClick = (button) => {
       }
 
       if (button.url_parms) {
-        buttonUrl += button.url_parms;
+        buttonUrl += ['?', getParams(button.url_parms, !!button.send_checked)].join('');
```

Comment:
There's some similarity on the line 116, maybe this could be avoided if you trigger the right branch and the stuff belows takes care of everything.

Reply:
@skateman i did look into that, in my case when i pass in `button.url` it will go thru the branch on line 58 and will always return on line 78, for non-explorer toolbar buttons we need the buttons to send up a get request which is why i need `button.url` to start with `/`. It looks like we do need the line i added in this branch.

Reply:
Why this though? As far as I can tell, this will only add `miq_grid_checks`, but the url is already right, isn't it?

Reply:
@himdel without this change, checked check box ids are not sent up with transaction, transaction looks like `chargeback_rate/edit#main_div`

Reply:
@himdel @skateman please re-review, my latest commit removes the above line but i had to make changes in `find_checked_items` [method](https://github.com/ManageIQ/manageiq-ui-classic/pull/7099/files#diff-6f28c7ea27bbb4811f9b8667d388d442L21) to get this working, I don't see how we could pass in `check_1` as params when sending up selected checkboxes from grid maybe that code branch has been broken for a while, any thoughts?

Reply:
If there's no `url_parms`, it should not add `#main_div`, it should add `ManageIQ.record.recordId`. 

The url should be `/edit/123`, and sending checked items should not be needed since the button only works for 1 item.

What `find_checked_items` should handle: `check_123: "1"` - id is 123,
`miq_grid_checks: ["123"]``, or just `params[:id] = 123` which is what we're going for.

(Pretty sure the `check_...` ids are still used in 29 places, so can't change that part either.)

Reply:
(this is where it should go through: https://github.com/ManageIQ/manageiq-ui-classic/blob/master/app/javascript/components/miq-toolbar.jsx#L58-L65 )

Reply:
> (this is where it should go through: https://github.com/ManageIQ/manageiq-ui-classic/blob/master/app/javascript/components/miq-toolbar.jsx#L58-L65 )

in case of of list view `ManageIQ.record.recordId` will not be there.

Reply:
Ah, true, I was looking only at the singular toolbar, sorry.

Yeah, for plural, we probably still need to rely on `send_checked` sending `miq_grid_checks`.

Reply:
> Ah, true, I was looking only at the singular toolbar, sorry.
> 
> Yeah, for plural, we probably still need to rely on `send_checked` sending `miq_grid_checks`. (or work around it by adding a function to compute the url?)

best solution and way i could get this working is to send in `send_checked`, `url_params` and keep the change in miq_toolbar.jsx that is `buttonUrl += ['?', getParams(button.url_parms, !!button.send_checked)].join('');` basically revert my last commit.

Reply:
:+1: from me, there really doesn't seem any other way of using send_checked + url yet

Reply:
@himdel @skateman bump

Reply:
I'm okay with it.

Reply:
This does change the behaviour for combinations of `url == '/foo'` (starts with slash) and `'url_parms == "main_div"` (doesn't start with question mark), and allows for serialized form data to end up in URL if ever used that way.

..but if it were ever used that way the existing behaviour would create invalid URLs, so I think we're safe :).